DigitalLearningSolutions.Data/Helpers/FilteringHelper.cs

```diff
@@ -39,19 +39,20 @@ public static string BuildFilterValueString(string group, string propertyName, s
         public static string? GetFilterString(
             string? existingFilterString,
             string? newFilterToAdd,
+            bool clearFilters,
             HttpRequest request,
             string cookieName,
             string? defaultFilterValue = null
         )
         {
-            if (existingFilterString == null && newFilterToAdd == null)
+            if (existingFilterString == null && newFilterToAdd == null && !clearFilters)
             {
                 return request.Cookies.ContainsKey(cookieName)
-                    ? request.Cookies[cookieName]
+                    ? request.Cookies[cookieName] == EmptyFiltersCookieValue ? null : request.Cookies[cookieName]
                     : defaultFilterValue;

```

Comment:
Slightly hard to understand what's going on here. Could we explain it a bit more by pulling out predicates as variables we can give helpful names to?
E.g. `cookieHasBeenSet = request.Cookies.ContainsKey(cookieName)` and `noFilterRequestedInQueryParams = existingFilterString == null && newFilterToAdd == null`